src/ml_sumcheck/protocol/prover.rs

```diff
@@ -79,9 +81,13 @@ impl<F: Field> IPForMLSumcheck<F> {
             // fix argument
             let i = prover_state.round;
             let r = prover_state.randomness[i - 1];
-            for multiplicand in prover_state.flattened_ml_extensions.iter_mut() {
+            #[cfg(not(feature = "parallel"))]
+            let iter_mut = prover_state.flattened_ml_extensions.iter_mut();
+            #[cfg(feature = "parallel")]
+            let iter_mut = prover_state.flattened_ml_extensions.par_iter_mut();
+            iter_mut.for_each(|multiplicand| {
```

Comment:
We have a macro [`ark_std::cfg_iter_mut`](https://docs.rs/ark-std/latest/ark_std/macro.cfg_iter_mut.html) that allows avoiding boilerplate like this.

Reply:
Oh, nice. Good to know.

Reply:
I updated it. Unfortunately, I can't use it for the larger change because `fold`/`reduce` work different with `rayon`.

Reply:
Right, for `fold` and `reduce` I have a different strategy: 

```rust
#[cfg(feature = "parallel")]
let init = || init_value;

#[cfg(not(feature = "parallel"))]
let init = init_value;

cfg_iter!(your_iterator).reduce(init, || { /* your code */});
```

Perhaps a nice improvement in the future would be to add a `cfg_reduce_init` macro to `ark_std` that can be used to set `init` based on `feature = "parallel"`.

Reply:
Something like this. I feel like it might be kind of obfuscating though, and it's a bit more unwieldy with fold.
```rust
#[macro_export]
macro_rules! cfg_reduce_init {
    ($e: expr, $init: expr, $op: expr) => {{
        #[cfg(feature = "parallel")]
        let result = $e.reduce(|| $init, $op);

        #[cfg(not(feature = "parallel"))]
        let result = $e.fold($init, $op);

        result
    }};
}
```